src/generators.rs

```diff
@@ -70,29 +79,31 @@ pub fn directed_cycle_graph(
     bidirectional: bool,
     multigraph: bool,
 ) -> PyResult<digraph::PyDiGraph> {
-    let mut graph = StablePyGraph::<Directed>::default();
     if weights.is_none() && num_nodes.is_none() {
         return Err(PyIndexError::new_err(
             "num_nodes and weights list not specified",
         ));
     }
-    let node_len: usize;
+    let node_len = get_num_nodes(&num_nodes, &weights);
+    let num_edges = if bidirectional {
+        2 * node_len
+    } else {
+        node_len
+    };
+    let mut graph = StablePyGraph::<Directed>::with_capacity(node_len, num_edges);
+
     let nodes: Vec<NodeIndex> = match weights {
         Some(weights) => {
             let mut node_list: Vec<NodeIndex> = Vec::new();
-            node_len = weights.len();
             for weight in weights {
                 let index = graph.add_node(weight);
                 node_list.push(index);
             }
             node_list
         }
-        None => {
-            node_len = num_nodes.unwrap();
-            (0..num_nodes.unwrap())
-                .map(|_| graph.add_node(py.None()))
-                .collect()
-        }
+        None => (0..num_nodes.unwrap())
```

Comment:
We have already unwrapped this value (same comment in other places too)
```suggestion
        None => (0..node_len)
```

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/625/commits/06c4442e7acb6d63ed37ce0510df9abf2122453f